### PR TITLE
Improving CONTRIBUTING guide

### PR DESCRIPTION
**Pull request by atilag**
TODO:
- [x] Debug builds
- [x] How to debug Python extensions

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -66,27 +65,45 @@ Most of the required dependencies can be installed via *pip*, using the
...
+   qiskit-aer$ python ./setup.py bdist_wheel
 
-    qiskit-aer$ python ./setup.py bdist_wheel
+
+This is also the way we will choose to change default `Aer` behavior by passing parameters to the build system.
+
+
+**Advanced options**
+
+As `Aer` is meant to be executed in many configurations and platforms, there is a complex underlaying build system that offers a lot of options you can tune by setting some parameters.
```
Comment (vvilpas) [inline CONTRIBUTING.md]: underlaying --> underlying

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -150,51 +178,51 @@ You further need to have *Xcode Command Line Tools* installed on macOS:
 
     $ xcode-select --install
 
-There are two ways of building Aer simulators, depending on your goal:
+There are two ways of building `Aer` simulators, depending on your goal:
 
 1. Build a Terra compatible add-on;
```
Comment (vvilpas) [inline CONTRIBUTING.md]: Terra compatible add-on --> python extension that works with Terra.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -150,51 +178,51 @@ You further need to have *Xcode Command Line Tools* installed on macOS:
...
+   qiskit-aer$ python ./setup.py bdist_wheel
 
-For the former, we just need to call the `setup.py` script:
 
-    qiskit-aer$ python ./setup.py bdist_wheel
+This is also the way we will choose to change default `Aer` behavior by passing parameters to the build system.
+
+***Advanced options***
+
+As `Aer` is meant to be executed in many configurations and platforms, there is a complex underlaying build system that offers a lot of options you can tune by setting some parameters.
```
Comment (vvilpas) [inline CONTRIBUTING.md]: underlaying --> underlying

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -216,30 +244,123 @@ option):
...
+
+
+   (QiskitDevEnv) qiskit-aer > python ./setup.py bdist_wheel
+
+
+This is also the way we will choose to change default `Aer` behavior by passing parameters to the build system.
+
+***Advanced options***
+
+As `Aer` is meant to be executed in many configurations and platforms, there is a complex underlaying build system that offers a lot of options you can tune by setting some parameters.
```
Comment (vvilpas) [inline CONTRIBUTING.md]: underlaying --> underlying

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -248,103 +369,150 @@ qiskit-aer$ python ./setup.py bdist_wheel -- -DUSEFUL_FLAG=Value
...
-AER_THRUST_BACKEND
-    We use Thrust library for GPU support through CUDA. If we want to build a version of Aer with GPU acceleration, we need to install CUDA and set this variable to the value: "CUDA".
+* AER_THRUST_BACKEND
+
+    We use Thrust library for GPU support through CUDA. If we want to build a version of `Aer` with GPU acceleration, we need to install CUDA and set this variable to the value: "CUDA".
     There are other values that will use different CPU methods depending on the kind of backend we want to use:
     - "OMP": For OpenMP support
     - "TBB": For Intel Threading Building Blocks
 
     Values: CUDA|OMP|TTB
```
Comment (vvilpas) [inline CONTRIBUTING.md]: TTB --> TBB

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -248,103 +369,150 @@ qiskit-aer$ python ./setup.py bdist_wheel -- -DUSEFUL_FLAG=Value
...
-    Example: ``cmake -DUSER_LIB_PATH=C:\path\to\openblas\libopenblas.so ..``
+    Example: ``python ./setup.py bdist_wheel -- -DUSER_LIB_PATH=C:\path\to\openblas\libopenblas.so``
+
+* BLAS_LIB_PATH
 
-BLAS_LIB_PATH
     Tells CMake the directory to look for the BLAS library instead of the usual paths.
     If no BLAS library is found under that directory, CMake will raise an error and stop.
 
     Values: An absolute path with file included.
```
Comment (vvilpas) [inline CONTRIBUTING.md]: An absolute path with file included. --> An absolute path.

Comment (atilag) [reply CONTRIBUTING.md]: xDDD

**Pull request merged**